src/main/java/seedu/address/model/project/Repository.java

```diff
@@ -16,9 +26,17 @@ public class Repository {
      */
     public Repository(String repository) {
         requireNonNull(repository);
+        checkArgument(isValidRepository(repository), MESSAGE_CONSTRAINTS);
         this.projectRepository = repository;
     }
 
+    /**
+     * Returns true if a given string is a valid repository.
+     */
+    public static boolean isValidRepository(String repository) {
+        return repository.matches(VALIDATION_REGEX);
+    }
+
     @Override
```

Comment:
Is it possible to make use of the toString method to display the repository as a github URL? an alternative would be to create another getter `getRepositoryUrl`